src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -54,17 +72,16 @@ def create_binding_constraint(
             properties=properties,
             terms=terms,
         )
-        if constraint.id in self.binding_constraints:
+        constraint.properties = constraint.local_properties.yield_binding_constraint_properties()
+
+        current_ini_content = self.ini_file.ini_dict_binding_constraints or {}
+        if any(values.get("name") == name for values in current_ini_content.values()):
```

Comment:
As I said in my first review we should check ids and not name

Reply:
Not resolved :/. We should check the ids and for that introduce a static method inside the class BindingConstraint to get the method use to calculate an id based on a name